src/Sarif.Viewer.VisualStudio/ErrorListService.cs

```diff
@@ -0,0 +1,155 @@
+            {
+                Instance.WriteRunLogToErrorList(runLog);
+            }
+
+            SarifTableDataSource.Instance.BringToFront();
+        }
+
+        private ErrorListService()
+        {
+            this.documentToLineIndexMap = new Dictionary<string, NewLineIndex>();
+        }
+
+        private Dictionary<string, NewLineIndex> documentToLineIndexMap;
+
+        private IRuleDescriptor GetRule(RunLog runLog, string ruleId)
+        {
+            if (runLog.RuleInfo == null)
+            {
+                return null;
+            }
+
+            foreach (IRuleDescriptor ruleDescriptor in runLog.RuleInfo)
+            {
+                if (ruleDescriptor.Id == ruleId) { return ruleDescriptor; }
+            }
+
+            throw new InvalidOperationException();
+        }
+
+        private void WriteRunLogToErrorList(RunLog runLog)
+        {
+            List<SarifError> sarifErrors = new List<SarifError>();
+
+            // Prefer optional fullName,  fall back to required Name property
+            string toolName = runLog.ToolInfo.FullName ?? runLog.ToolInfo.Name;
+
+            foreach (Result result in runLog.Results)
+            {
+                string message;
+                string document;
+                Region region;
+                NewLineIndex newLineIndex;
+
+                foreach (Location location in result.Locations)
+                {
+                    PhysicalLocationComponent physicalLocation;
+                    if (location.ResultFile != null)
+                    {
+                        physicalLocation = location.ResultFile[0];
+                    }
+                    else
+                    {
+                        physicalLocation = location.AnalysisTarget[0];
+                    }
+
+                    // TODO helper to provide best representation of URI
+                    document = physicalLocation.Uri.LocalPath;
+
+                    // TODO retrieve file from nested component
+                    region = physicalLocation.Region;
+
+                    IRuleDescriptor rule = GetRule(runLog, result.RuleId);
+                    message = result.GetMessageText(rule, concise: true);
+
+                    SarifError sarifError = new SarifError(document)
+                    {
+                        Region = region,
+                        ErrorCode = result.RuleId,
+                        IsError = IsError(result.Kind),
+                        Message = message,
+                        MimeType = physicalLocation.MimeType,
+                        Tool = toolName
+                    };
+
+                    if (region != null)
+                    {
+                        sarifError.ColumnNumber = region.StartColumn - 1;
+                        sarifError.LineNumber = region.StartLine - 1;
+                    }
+
+                    sarifErrors.Add(sarifError);
+                }
+
+                if (result.RelatedLocations != null)
```

Comment:
Do we want to do this? The canonical example is "Error: foo.js(20, 10): Used of tainted data" with related location "(5, 10): Data entered the system here". Do we want that to appear in the error list (and cause the error count to increase)?